apps/ejabberd/src/mongoose_admin.erl

```diff
@@ -0,0 +1,264 @@
+            {args, [{caller, binary}, {newpass, binary}]},
+            {result, ok}
+        ]
+    ].
+
+%%get_user_resources(Jid) ->
+%%    J = jid:from_binary(Jid),
+%%    ejabberd_sm:get_user_resources(J#jid.user, J#jid.server).
+
+
+kick_this_session(Jid) ->
+    J = jid:from_binary(Jid),
+    ejabberd_sm:route(
+        jid:make(<<"">>, <<"">>, <<"">>),
+        J,
+        {broadcast, {exit, <<"kicked">>}}),
+    <<"kicked">>.
+
+mk(E) ->
+    Ss = element(1, E),
+    {U, S, R} = Ss,
+    <<U/binary, "@", S/binary, "/", R/binary>>.
+
+listsessions(Host) ->
+    Lst = ejabberd_sm:get_vh_session_list(Host),
+    lists:map(fun mk/1, Lst).
+
+
+registered_users(Host) ->
+    Users = ejabberd_auth:get_vh_registered_users(Host),
+    SUsers = lists:sort(Users),
+    lists:map(fun({U, S}) -> <<U/binary, "@", S/binary>> end, SUsers).
+
+
+register(User, Host, Password) ->
+    case ejabberd_auth:try_register(User, Host, Password) of
+        ok ->
+            list_to_binary(io_lib:format("User ~s@~s successfully registered", [User, Host]));
+        {error, exists} ->
+            String = io_lib:format("User ~s@~s already registered at node ~p",
+                [User, Host, node()]),
+            throw({exists, String});
+        {error, Reason} ->
+            String = io_lib:format("Can't register user ~s@~s at node ~p: ~p",
+                [User, Host, node(), Reason]),
+            throw({error, String})
+    end.
+
+unregister(Jid) ->
+    J = jid:from_binary(Jid),
+    ejabberd_auth:remove_user(J#jid.luser, J#jid.lserver),
+    <<"">>.
+
+
+send_message(From, To, Body) ->
+    Packet = build_packet(message_chat, Body),
+    F = jid:from_binary(From),
+    T = jid:from_binary(To),
+    ejabberd_hooks:run(user_send_packet,
+        F#jid.lserver,
+        [F, T, Packet]),
+    % privacy check is missing, but is it needed?
+    ejabberd_router:route(F, T, Packet),
+    <<"">>.
+
+
+registered_commands() ->
+    [#{
+        name => mongoose_commands:name(C),
+        category => mongoose_commands:category(C),
+        action => mongoose_commands:action(C),
+        desc => to_binary(mongoose_commands:desc(C))
+    } || C <- mongoose_commands:list(admin)].
+
+
+get_recent_messages(Caller, Other, Limit) ->
+    Res = lookup_recent_messages(Caller, Other, Limit),
+    lists:map(fun record_to_map/1, Res).
+
+
+change_user_password(User, Password) ->
+    Jid = jid:from_binary(User),
+    ejabberd_auth:set_password(Jid#jid.luser, Jid#jid.lserver, Password).
+
+
+record_to_map({Id, From, Msg}) ->
+    Jbin = jid:to_binary(From),
+    {Msec, _} = mod_mam_utils:decode_compact_uuid(Id),
+    MsgId = case xml:get_tag_attr(<<"id">>, Msg) of
+                {value, MId} -> MId;
+                false -> <<"">>
+            end,
+    Body = exml_query:path(Msg, [{element, <<"body">>}, cdata]),
+    #{sender => Jbin, timestamp => round(Msec / 1000000), message_id => MsgId, body => Body}.
+
+build_packet(message_chat, Body) ->
```

Comment:
What are the other packets which could be built by this function? If we don't have them now, let's remove the first parameter and make the function simpler.